pydantic/_internal/_model_construction.py

```diff
@@ -384,6 +405,9 @@ def inspect_namespace(  # noqa C901
             and ann_type not in all_ignored_types
             and getattr(ann_type, '__module__', None) != 'functools'
         ):
+            if isinstance(ann_type, typing._AnnotatedAlias) and isinstance(ann_type.__metadata__[0], ModelPrivateAttr):
```

Comment:
I think you can use `pydantic._internal._typing_extra.is_annotated` to check if the annotation is `Annotated`